src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,195 @@
+        if not isinstance(interval, (int, float)):
+            raise TypeError(  # noqa: TRY003
+                "System monitoring `interval` should be an `int` or a `float`, but got "
+                f"'{type(interval)}'"
+            )
+        if not isinstance(num_samples, int):
+            raise TypeError(  # noqa: TRY003
+                "System monitoring `num_samples` should be an `int`, but got "
+                f"'{type(num_samples)}'"
+            )
+        if not isinstance(plot, bool):
+            raise TypeError(  # noqa: TRY003
+                f"System monitoring `plot` should be a `bool`, but got '{type(plot)}'"
+            )
+
+        self._interval = interval  # seconds
+        self._nb_samples = num_samples
+        self._plot = plot
+        self._warn_user = True
+
+    def __call__(self, live):
+        self._live = live
+        self._shutdown_event = Event()
+        Thread(
+            target=self._monitoring_loop,
+        ).start()
+
+    def _monitoring_loop(self):
+        while not self._shutdown_event.is_set():
+            self._metrics = {}
+            for _ in range(self._nb_samples):
+                last_metrics = {}
+                try:
+                    last_metrics = self._get_metrics()
+                except psutil.Error:
+                    if self._warn_user:
+                        logger.exception("Failed to monitor CPU metrics")
+                        self._warn_user = False
+
+                self._metrics = merge_with(sum, self._metrics, last_metrics)
+                self._shutdown_event.wait(self._interval)
+                if self._shutdown_event.is_set():
+                    break
+            for name, values in self._metrics.items():
+                blacklisted = any(
+                    name.startswith(prefix) for prefix in self._plot_blacklist_prefix
+                )
+                self._live.log_metric(
+                    name,
+                    values / self._nb_samples,
+                    timestamp=True,
+                    plot=None if blacklisted else self._plot,
+                )
+
+    @abc.abstractmethod
+    def _get_metrics(self) -> Dict[str, Union[float, int]]:
+        pass
+
+    def end(self):
+        self._shutdown_event.set()
+
+
+class CPUMonitor(_SystemMonitor):
+    _plot_blacklist_prefix: Tuple = (
+        METRIC_CPU_COUNT,
+        METRIC_RAM_TOTAL_GB,
+        "system/disk/total (GB)",
+    )
+
+    def __init__(
+        self,
+        interval: float = 0.5,
+        num_samples: int = 10,
+        disks_to_monitor: Optional[Dict[str, str]] = None,
+        plot: bool = True,
+    ):
+        """Monitor CPU resources and log them to DVC Live.
+
+        Args:
+            interval (float): interval in seconds between two measurements.
+                Defaults to 0.5.
+            num_samples (int): number of samples to average. Defaults to 10.
+            disks_to_monitor (Optional[Dict[str, str]]): paths to the disks or
```

Comment:
I'm not sure I understand this "aths to the disks or partitions to monitor disk usage statistics.". what is the difference disk vs parittion?

Reply:
I tried to change the description for this arg. Let me know what you think. I mentioned `disk` because I'm afraid our users don't know what a partition is. 

Changes involve:
 - renaming to `folders_to_monitor`. While not perfectly accurate (because two folders on the same partition, that have different content, will have identical statistics with this function), it is the most understandable name I could find. I'm scared that `partition_to_monitor` doesn't speak to our users. 
 - mentioning the perfectly accurate definition in the rest of the docstring so people looking for deeper information find what they need.